### PR TITLE
ref: Preserve project state if it did not change

### PR DESCRIPTION
**Pull request by untitaker**
See code comments for explanation. I had to move `last_fetch` out because I am not able to bump it once `ProjectState` is in an arc

### CODE REVIEW

**Review by mitsuhiko — APPROVED** [review]
AMEN

File: relay-server/src/actors/project.rs [hunk 1]
```diff
@@ -557,11 +587,35 @@ impl Project {
...
+                            slf.state = Some(response.state.clone());
+                            response.state
+                        };
+
+                        slf.last_fetch = Utc::now();
+                        sender.send(send_state).ok();
+                    }
+                    Err(()) => {
+                        // should be unreachable
+                        log::error!("received unit-error from FetchProjectState");
```
Comment (jan-auer) [inline relay-server/src/actors/project.rs:616]: This is not unreachable, and the error condition is already logged.

Comment (jan-auer) [reply relay-server/src/actors/project.rs:616]: Or not, looks like we might swallow the errors in some cases.

Comment (untitaker) [reply relay-server/src/actors/project.rs:616]: The one I was going off of was https://github.com/getsentry/relay/blob/b58ce9e707e92e1ff8540e90ed8c911452afb1c5/relay-server/src/actors/project_cache.rs#L212... so a few mailbox errors bubble up like this. It's not a very descriptive error message but I would split it up once it happens

**Pull request merged**